pages/languages/go.md

```diff
@@ -3,3 +3,334 @@ permalink: /languages/go/
+Starting with the 1.5 series, [Go must be compiled using an earlier version of
+Go](https://docs.google.com/document/d/1OaatvGhEAq7VseQ9kkavxKNAfepWy2yhPUBs96FGV28/edit).
+To install the latest Go version from the source distribution (since [the Go
+1.5 binary install isn’t currently working on OS
+X](https://github.com/moovweb/gvm/issues/152)), start by installing the
+latest 1.4.x version (1.4.3 at the time of writing):
+
+```sh
+$ gvm install go1.4.3
+$ gvm use go1.4.3
+```
+
+Until `gvm` v1.0.23 is released, which will integrate
+[moovweb/gvm#154](https://github.com/moovweb/gvm/pull/154) (aka
+[e34bbfc](https://github.com/moovweb/gvm/commit/e34bbfc017c22c194564a5f9cc3c83d0aa6c1b03)),
+explicitly setting `GOROOT_BOOTSTRAP` is necessary when invoking `gvm
+install`:
+
+```sh
+$ GOROOT_BOOTSTRAP=$GOROOT gvm install go1.5.1
+$ gvm use go1.5.1 --default
+```
+
+## <a name="gopath"></a>Understanding workspaces and GOPATH
+
+Much of the power and convenience of the Go programming environment comes from
+its strongly-opinionated concept of workspace layout. The official [How to
+Write Go Code](https://golang.org/doc/code.html) document contains an
+excellent explanation and walkthrough, but the highlights are:
+
+- The `GOPATH` environment variable identifies the root directory of all Go
+  workspaces.
+- The source code for packages is installed under `$GOPATH/src`.
+- New workspaces should be created under `$GOPATH/src`.
+- The convention for package paths under `$GOPATH/src` is
+  `repository_server/user_or_org/package_or_project_name`.
+  - e.g: `github.com/18F/hmacauth`
+- This same package path convention is used to install packages using
+  [`go get`](go-get) and to import packages into Go programs via the [`import
+  statement`](https://golang.org/doc/effective_go.html#names).
+
+## <a name="go-get"></a>Package management with go get
+
+The `go get` command is the standard tool for installing Go packages and
+projects into `$GOPATH`. It relies upon the conventions described above to
+fetch and install open source packages. For example, using the real-world
+[18F/hmacauth](https://github.com/18F/hmacauth/) package:
+
+```sh
+$ go get github.com/18F/hmacauth
+```
+
+will install the package under `$GOPATH/src/github.com/18F/hmacauth` by
+performing a `git clone` on the original source repository. That is to say,
+the package directory will contain a local Git repository that can be used for
+local development. The package can then be imported into a Go program via:
+
+```go
+import "github.com/18F/hmacauth"
+```
+
+`go get` will also automatically scan the Go source files within the target
+package or project and automatically install any other Go package
+dependencies. A later section of this document will go into [Go dependency
+management](#dependencies) in detail.
+
+## go get and gvm
+
+`gvm use VERSION` will set `GOPATH` to a version-specific directory, and will
+automatically add `$GOPATH/bin` to your `PATH`. This means you may have to
```

Comment:
I like this, so I'm going to add a similar thing to the JS guide since `nvm` (and I presume `n` work in the same way)